### PR TITLE
Fix Redis custom codec registration when there is more than one codec

### LINKED ISSUE #35661: Custom redis codecs are not registered when more than one Codec is implemented
### Describe the bug

According to the documentation of the redis-client extenstion, custom codecs can be registered, see https://quarkus.io/guides/redis-reference#custom-codec .

However when more than one bean implementing the `Codec` interface is present, no `Codec` is registered.

The problem seems to be the in the `RedisClientRecorder` class:
https://github.com/quarkusio/quarkus/blob/45d914a2edabd1b9c1976c0435e9fbc8e19b77f6/extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/client/RedisClientRecorder.java#L68

The `codecs.isResolvable()` checks if there is exactly one implementation of a `Codec` resolvable, so if more than one codec is resolvable no codec is registered.

### Expected behavior

All custom codecs should be registered.

### Actual behavior

If more than one bean implements the `Codec` interface, no custom codec is registered.

### How to Reproduce?

Create two beans implementing the `Codec` interface and check if the codec is registered.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.3.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by domkun**
Removes `isResolvable()` check and adds a second codec to the custom codec test.

Fixes #35661

### CODE REVIEW

File: extensions/redis-client/deployment/src/test/java/io/quarkus/redis/deployment/client/datasource/CustomCodecTest.java [hunk 1]
```diff
@@ -89,4 +129,14 @@ public Jedi(String firstName, String lastName) {
         }
     }
 
+    public static class Sith {
```
Comment (cescoffier) [inline extensions/redis-client/deployment/src/test/java/io/quarkus/redis/deployment/client/datasource/CustomCodecTest.java:132]: Love it!

**Review by cescoffier — APPROVED** [review]
Thanks!

**Comment by domkun** [comment]
Damn, forgot to push my copy-paste mistake. Will update the PR soon. Sorry.

**Comment by cescoffier** [comment]
Can you squash your commit to have a single one?

**Comment by domkun** [comment]
Sure, commits are squashed now.

**Review by geoand — APPROVED** [review]

**Comment by aloubyansky** [comment]
This does not appear to be a complete change for 3.2 backport. This change adjusts implementation of `RedisClientRecorder._registerCodecs()` which isn't present in 3.2 at all.

**Comment by cescoffier** [comment]
Yeah, I would not backport it. It would require too many changes.

**Comment by aloubyansky** [comment]
Thanks for the info, removed from 3.2 backports for now.

**Pull request merged**